Development build problems

<!-- The text within this markup is a comment, and is intended to provide
guidelines to open an issue for the ITK repository. This text will not
be part of the issue. -->


<!-- Before submitting an issue, please check that your issue has not been
already filed. -->

### Description
Development ITK fails to build for me.  There are numerous errors of this sort:
/home/melrobin/packages/ITK/Modules/Filtering/ImageFilterBase/include/itkCastImageFilter.hxx:93:73:   required from ‘void itk::CastImageFilter<TInputImage, TOutputImage>::DynamicThreadedGenerateData(const OutputImageRegionType&) [with TInputImage = itk::Image<itk::RGBPixel<unsigned char> >; TOutputImage = itk::Image<itk::Vector<float, 3> >; itk::CastImageFilter<TInputImage, TOutputImage>::OutputImageRegionType = itk::ImageRegion<2>]’
/home/melrobin/packages/ITK/Modules/Filtering/ImageFilterBase/test/itkCastImageFilterTest.cxx:53:21:   required from here
/home/melrobin/packages/ITK/Modules/Core/Common/include/itkMetaProgrammingLibrary.h:183:36: error: no matching function for call to ‘itk::Vector<float, 3>::Vector(itk::RGBPixel<unsigned char>)’
                           decltype(static_cast<TToType>(std::declval<TFromType>()))>
                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


### Steps to Reproduce
The error comes when building this object:
Building CXX object Modules/Filtering/ImageFilterBase/test/CMakeFiles/ITKImageFilterBaseTestDriver.dir/itkCastImageFilterTest.cxx.o



### Expected behavior

Completed build

### Actual behavior

Build Error

### Reproducibility



### Versions
Development

### Environment

Fedora 29, gcc 8.3.1

### Additional Information

<!-- Any additional information, configuration or data that might be necessary
to reproduce the issue. -->


<!-- **Note**: Use issues for their purpose; issues are not for code help. Need
help? Ask your question at ITK discourse (https://discourse.itk.org/). -->

FYI, one my old module fails to build with current master branch too

```
Modules/Core/Common/include/itkMetaProgrammingLibrary.h:183:36: error:
no matching function for call to
‘itk::Vector<float, 3>::Vector(itk::RGBPixel<short int>)’
decltype(static_cast<TToType>(std::declval<TFromType>()))>
```


```
Modules/Core/Common/include/itkImageAlgorithm.hxx:45:17: error:
no matching function for call to
‘itk::Vector<float, 3>::Vector(itk::ImageConstIterator<itk::Image<itk::RGBPixel<short int>, 3> >::PixelType)’
ot.Set( static_cast< typename OutputImageType::PixelType >( it.Get() ) );
```

Best regards

--- a/Modules/Core/Common/include/itkMetaProgrammingLibrary.h
+++ b/Modules/Core/Common/include/itkMetaProgrammingLibrary.h
@@ -215,15 +215,10 @@ struct Not : NotC<TF::Value>
  *
  * Identifies if "static_cast<TToType>(TFromType)" can be done.
  */
-template <typename TFromType, typename TToType, typename TNoDeclType = TToType>
-struct is_static_castable : std::false_type
-{};
-/// \cond SPECIALIZATION_IMPLEMENTATION
 template <typename TFromType, typename TToType>
-struct is_static_castable<TFromType, TToType, decltype(static_cast<TToType>(std::declval<TFromType>()))>
-  : std::true_type
-{};
-/// \endcond
+using is_static_castable = std::integral_constant<bool,
+                                                  std::is_constructible<TToType, TFromType>::value ||
+                                                    std::is_convertible<TFromType, TToType>::value>;
 } // namespace mpl
 
 

